docsource/ops.rst

```diff
@@ -1,6 +1,228 @@
+
+.. code-block:: bash
+
+    POST _reindex
+    {
+      "source": {
+        "index": "kbase.genome_1"
+      },
+      "dest": {
+        "index": "kbase.genome_1_r1",
+        "version_type": "external"
+      },
+      "script": {
+        "lang": "painless",
+        "inline": "if (ctx._source.foo == 'bar') {ctx._version++; ctx._source.remove('foo')}"
+      }
+    }
+
+6. Now, reindex the entire data from current index to new index. Alternately, use a query to reindex only a subset of the current index.
+
+.. code-block:: bash
+
+    POST _reindex
+    {
+      "source": {
+        "index": "kbase.genome_1"
+      },
+      "dest": {
+        "index": "kbase.genome_1_r1"
+      }
+    }
+
+        OR
+
+    POST _reindex
+    {
+      "source": {
+        "index": "kbase.genome_1",
+        "query": {
+          ...
+        }
+      },
+      "dest": {
+        "index": "kbase.genome_1_r1"
+      }
+    }
+
+7. Run a checksum on the new index to make sure the numbers line up with the numbers of the current index.
+
+.. code-block:: bash
+
+    GET /kbase.genome_1_r1/_stats/docs,store
+
+8. Run a query to specifically check the change that was applied.
+
+.. code-block:: bash
+
+    GET kbase.genome_1_r1/_search
+
+   OR
+
+    GET kbase.genome_1_r1/_search
+    {
+     "query": {
+       "match": {
+         "FIELD": "VALUE"
+       }
+     }
+    }
+
+   OR
+
+    https://www.elastic.co/guide/en/elasticsearch/reference/5.5/search-request-body.html
+
+9. If the new index looks good, update index alias and delete current index.
+
+.. note::
+
+    If you want the current index to linger for a day or two to serve a rollback option, reindex the current index into another new index called kbase.genome_1_backup and then delete the current index. This is one of two ways of renaming an index in ElasticSearch. The other way is to use the snapshot API.
+
+.. code-block:: bash
+
+    POST _aliases
+    {
+     "actions": [
+     {
+       "add": {
+         "index": "kbase.genome_1_r1",
+         "alias": "kbase.genome"
+         }
+       },
+       {
+         "remove": {
+         "index": "kbase.genome_1",
+         "alias": "kbase.genome"
+       }
+     }
+     ]
+    }
+
+    DELETE kbase.genome_1
+
+10. List all available indexes for the genome alias and all available genome indexes to ensure consistency across the alias map. Verify that all genome indexes that are present (except for backups) are referenced by the alias. Also verify that the alias does not contain an index reference for which no index exists.
+
+.. code-block:: bash
+
+    GET /_cat/aliases/kbase.genome
+
+    GET /_cat/indices/kbase.genome_*
+
+11. If the change involved in the reindexing operation also requires a corresponding search type spec change (located in resources/types/genome.yml for example), then this change must be applied.
```

Comment:
applied to what? the spec?

Reply:
Yes. Added "...then this change must be applied to the spec as well.". Your idea about creating the index from spec will take care of this step too. I have left this for a future PR.